dev/reference/sql/data-types/date-and-time.md

```diff
@@ -230,3 +191,71 @@ When numeral `00` is inserted to `YEAR(4)`, the result is 0000 rather than 2000.
 If you want the result to be 2000, specify the value to be 2000, '0' or '00'.
 
 The two-digit year-portion might not be properly calculated in some functions such  `MIN()` and  `MAX()`. For these functions, the four-digit format suites better.
+
+## Supported types
+
+### `DATE` type
+
+`DATE` only contains date-portion and no time-portion, displayed in `YYYY-MM-DD` format. The supported range is '1000-01-01' to '9999-12-31':
+
+```sql
+DATE
+```
+
+### `TIME` type
+
+For the `TIME` type, the format is `HH:MM:SS[.fraction]` and valid values range from '-838:59:59.000000' to '838:59:59.000000'. `TIME` is used not only to indicate the time within a day but also to indicate the time interval between 2 events.  An optional `fsp` value in the range from 0 to 6 may be given to specify fractional seconds precision. If omitted, the default precision is 0:
+
+```sql
+TIME[(fsp)]
+```
+
+> **Note:**
+> 
+> Pay attention to the abbreviated form of `TIME`. For example, '11:12' means '11:12:00' instead of '00:11:12'. However, '1112' means '00:11:12'. These differences are caused by the presence or absence of the `:` character.
+
+### `DATETIME` type
+
+`DATETIME` contains both date-portion and time-portion. Valid values range from '1000-01-01 00:00:00.000000' to '9999-12-31 23:59:59.999999'.
+
+TiDB displays `DATETIME` values in `YYYY-MM-DD HH:MM:SS[.fraction]` format, but permits assignment of values to `DATETIME` columns using either strings or numbers.  An optional fsp value in the range from 0 to 6 may be given to specify fractional seconds precision. If omitted, the default precision is 0:
+
+```sql
+DATETIME[(fsp)]
+```
+
+### `TIMESTAMP` type
+
+`TIMESTAMP` contains both date-portion and time-portion. Valid values range from '1970-01-01 00:00:01.000000' to '2038-01-19 03:14:07.999999' in UTC time. An optional fsp value in the range from 0 to 6 may be given to specify fractional seconds precision. If omitted, the default precision is 0.
+
+In `TIMESTAMP`, zero is not permitted to appear in the month-portion or day-portion. The only exception is zero value itself '0000-00-00 00:00:00'.
+
+```sql
+TIMESTAMP[(fsp)]
+```
+
+#### Timezone Handling
+
+When `TIMESTAMP` is to be stored, TiDB converts the `TIMESTAMP` value from the current time zone to UTC time zone. When `TIMESTAMP`  is to be retrieved, TiDB converts the stored `TIMESTAMP` value from UTC time zone to the current time zone (Note: `DATETIME` is not handled in this way). The default time zone for each connection is the server's local time zone, which can be modified by the environment variable `time_zone`.
+
+> **Warning:**
+> 
+> As in MySQL, the `TIMESTAMP` data type suffers from the [Year 2038 Problem](https://en.wikipedia.org/wiki/Year_2038_problem). For storing values that may span beyond 2038, please consider using the `DATETIME` type instead.
```

Comment:
as far as know, TiDB doesn't use 32-bit integer to store timestamp, so there is no `Year 2038 problem` in TiDB.

Reply:
It uses 64-bit, but there are 3 type systems in TiKV (one for schema, one for comparisons and one for storage). The schema type system enforces range, and prevents values beyond 2038 from being stored:

```
mysql> CREATE TABLE t1 (a int not null primary key auto_increment, b timestamp not null);
Query OK, 0 rows affected (0.08 sec)

mysql> INSERT INTO t1 (b) VALUES ('2039-01-01 10:00:00');
ERROR 1292 (22007): invalid time format: '{2039 1 1 10 0 0 0}'
mysql> INSERT INTO t1 (b) VALUES ('2030-01-01 10:00:00');
Query OK, 1 row affected (0.01 sec)
```